src/sentry/static/sentry/app/utils/discover/discoverQuery.tsx

```diff
@@ -87,18 +100,42 @@ class DiscoverQuery extends React.Component<Props, State> {
     );
   };
 
-  fetchData = () => {
-    const {eventView, orgSlug, location, limit, keyTransactions} = this.props;
+  getRoute(keyTransactions, trendsType) {
+    if (keyTransactions) {
+      return 'key-transactions';
+    }
+    if (trendsType) {
+      return 'events-trends';
+    }
+    return 'eventsv2';
+  }
 
-    if (!eventView.isValid()) {
+  fetchData = () => {
+    const {
+      eventView,
+      orgSlug,
+      location,
+      limit,
+      keyTransactions,
+      trendChangeType,
+    } = this.props;
+
+    if (!eventView.isValid() && !trendChangeType) {
       return;
     }
 
-    const route = keyTransactions ? 'key-transactions' : 'eventsv2';
+    const route = this.getRoute(keyTransactions, trendChangeType);
 
     const url = `/organizations/${orgSlug}/${route}/`;
-    const tableFetchID = Symbol('tableFetchID');
-    const apiPayload = eventView.getEventsAPIPayload(location);
+    const tableFetchID = Symbol(`tableFetchID`);
+    const apiPayload: EventQuery &
```

Comment:
Minor, but any recommendations on fixing this part? I need to add `trendFunction` for the request and didn't want to use `any`